demos/simple-storage/repro-build-protocol/web3api.build.yaml

```diff
@@ -0,0 +1,9 @@
+image: 
+  name: build-env
+  dockerfile: ./myCustomDockerfile #Comment to use default image
```

Comment:
I think eventually we will want to support various images. I'm not sure if we want to detect which one should be used automatically or have the user specify an image that's not necessarily published by Web3API like `rust:1.40` which we then use a base layer and doing whatever we need on top of it. That all may be outside the scope of this initial pull request though.